### PR TITLE
Upgrade snap to use mysql version 8.0.37

### PR DESCRIPTION
**Pull request by shayancanonical**
## Issue
We can no longer re-build the snap because mysql versions 8.0.37 are available, and support for 8.0.36 is dropped from deb packages.

## Solution
Upgrade to use 8.0.37

### CODE REVIEW

**Comment by shayancanonical** [comment]
VM test pass in https://github.com/canonical/mysql-operator/pull/470

**Review by carlcsaposs-canonical — APPROVED** [review]

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -128,9 +128,9 @@ parts:
   packages-deb:
     plugin: nil
     stage-packages:
-      - mysql-server-8.0=8.0.36-0ubuntu0.22.04.1
-      - mysql-router=8.0.36-0ubuntu0.22.04.1
-      - mysql-shell=8.0.36+dfsg-0ubuntu0.22.04.1~ppa4
+      - mysql-server-8.0=8.0.37-0ubuntu0.22.04.3
+      - mysql-router=8.0.37-0ubuntu0.22.04.3
+      - mysql-shell=8.0.37+dfsg-0ubuntu0.22.04.1~ppa3
```
Comment (paulomach) [inline snap/snapcraft.yaml:133]: out of ignorance: what does the `ppa4 -> ppa3` means here?

**Review by paulomach — APPROVED** [review]

File: snap/snapcraft.yaml (hunk 1 above)
Comment (shayancanonical) [reply snap/snapcraft.yaml:133]: i am not sure to be honest, the ~ in a package name is used to indicate a version number or possibly a build. may be it indicates that `8.0.37` was built 3 times? this is a total guess though

https://launchpad.net/~data-platform/+archive/ubuntu/mysql-archive -> contains `8.0.36...~ppa4` where as I got the version number for 8.0.37 from https://launchpad.net/~data-platform/+archive/ubuntu/mysql-shell

Comment (taurus-forever) [reply snap/snapcraft.yaml:133]: It is the amount of changes/try to build PPA. Every time you dput the package you need to bump version to produce the next package. for 8.0.37 we were a bit more lucky (try #3 passed all tests) comparing to 8.0.36 (we build it 4 times). :-D

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 name: charmed-mysql # you probably want to 'snapcraft register <name>'
 base: core22 # the base snap is the execution environment for this snap
-version: '8.0.36' # just for humans, typically '1.2+git' or '1.3.2'
+version: '8.0.37' # just for humans, typically '1.2+git' or '1.3.2'
```
Comment (taurus-forever) [inline snap/snapcraft.yaml:3]: Consider to remove the comments here and above it the future. It was relevant example for the snap cration from template phase. :-D

**Review by taurus-forever — APPROVED** [review]

**Pull request merged**